libnavigation-core/src/test/java/com/mapbox/navigation/trip/service/MapboxTripServiceTest.kt

```diff
@@ -13,12 +15,25 @@ import org.junit.Test
 class MapboxTripServiceTest {
 
     private lateinit var service: MapboxTripService
-    private val notification: TripNotification = mockk()
+    private val tripNotification: TripNotification = mockk()
+    private val notification: Notification = mockk()
     private val callback: () -> Unit = { }
 
     @Before
     fun setUp() {
-        service = MapboxTripService(notification, callback)
+        service = MapboxTripService(tripNotification, callback)
+        every { tripNotification.getNotificationId() } answers { 1234 }
+        every { tripNotification.getNotification() } answers { notification }
+        every { tripNotification.onTripSessionStopped() } answers { Unit }
+
+    }
+
+    @Test
+    fun testServiceStartStop() {
```

Comment:
I also tried running this test on `master` and it doesn't reproduce any kind of issue.

We'll need a better test case that reproduces the issue or assert that the right coroutines API method is called.

Reply:
There is nothing to assert, because there is no result that we are interested in. The test makes sure that calling start()/stop() does not crash the app. I can rename the test.

Reply:
**Regarding the difference between job.cancel() and job.close():**
The difference between close() and cancel() is that cancel() will remove all messages from the channel before closing it. close() does not remove messages. If you have a client waiting for onReceive on a closed channel that client will receive data, but since the channel is closed, the client will throw an exception. So, always use cancel(). If you use close(), you’ll have to get try/catch involved. 
You should always cancel a static channel during shutdown. If the channel is not canceled it holds on to what ever objects it may contain, causing a leak. If the channel’s sope is not static, i.e. the channel is a member variable of a class, then you don’t need to cancel() it.

Reply:
I was thinking that this test is supposed to catch the regression. In this case, let's try to do this when writing all the other tests when covering the tailwork.